src/net_processing.cpp

```diff
@@ -3330,6 +3330,15 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         }
         if (!vRecv.empty())
             vRecv >> fRelay;
+
+        // Don't waste any of our full outbound slots on peers that don't want tx relay (because they might be in -blocksonly mode)
```

Comment:
In 82e7a7dc3b4447e4affe3a98e1af8b478aa4dc04: Perhaps we could mention on this comment that it doesn't apply in case we're running in `-blocksonly` mode.
```suggestion
        // Don't waste any of our full outbound slots on peers that don't want tx relay (because they might be in -blocksonly mode) unless we are in -blocksonly mode.
```

Reply:
I have included this info in the doc for the new spot!